packages/salesforcedx-vscode-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -178,6 +219,75 @@ export class ApexReplayDebug extends LoggingDebugSession {
     this.sendResponse(response);
   }
 
+  protected async scopesRequest(
+    response: DebugProtocol.ScopesResponse,
+    args: DebugProtocol.ScopesArguments
+  ): Promise<void> {
+    response.success = true;
+    const frameInfo = this.logContext.getFrameHandler().get(args.frameId);
+    if (!frameInfo) {
+      response.body = { scopes: [] };
+      this.sendResponse(response);
+      return;
+    }
+    const scopes = new Array<Scope>();
```

Comment:
Every scope we push needs a unique ID because when we enter scopesRequest this ID is what we will use to retrieve the ScopeContainer. Gunnar used the Handles class provided by the debugger api so I followed this as well.